File: UNXNavigatorDemo/Feature/Basic/Controllers/ViewController01_BackgroundColor.m
//
//  ViewController01_BackgroundColor.m
//  UNXNavigatorDemo
//
//  Created by Leo Lee on 2020/10/25.
//

#import <UNXNavigator/UNXNavigator.h>

#import "ViewController01_BackgroundColor.h"
#import "UIColor+RandomColor.h"

@interface ViewController01_BackgroundColor ()

@end

@implementation ViewController01_BackgroundColor

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationController.navigationBar.largeTitleTextAttributes = [self unx_titleTextAttributes];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    self.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeAutomatic;
    self.navigationController.navigationBar.prefersLargeTitles = YES;
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    self.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeNever;
    self.navigationController.navigationBar.prefersLargeTitles = NO;
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIColor *)unx_barTintColor {
    return [UIColor whiteColor];
}

- (NSDictionary<NSAttributedStringKey,id> *)unx_titleTextAttributes {
    return @{NSForegroundColorAttributeName: [self unx_barTintColor]};
}

- (UIColor *)unx_navigationBarBackgroundColor {
    return [UIColor customDarkGrayColor];
}

@end

File: Example/UNXNavigatorSwiftDemo/Basic/ViewController01_BackgroundColor.swift
//
//  ViewController01_BackgroundColor.swift
//  UNXNavigatorSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class ViewController01_BackgroundColor: BaseTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        navigationController?.navigationBar.largeTitleTextAttributes = unx_titleTextAttributes
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        navigationItem.largeTitleDisplayMode = .automatic
        navigationController?.navigationBar.prefersLargeTitles = true
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        navigationItem.largeTitleDisplayMode = .never
        navigationController?.navigationBar.prefersLargeTitles = false
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    override var unx_barTintColor: UIColor? {
        return .white
    }
    
    override var unx_titleTextAttributes: [NSAttributedString.Key : Any]? {
        return [NSAttributedString.Key.foregroundColor: UIColor.white]
    }

    override var unx_navigationBarBackgroundColor: UIColor? {
        return .customDarkGray
    }
}
